### ISSUE TITLE
Cannot read property 'saveAll' of undefined (ee.Join.saveAll)

### ISSUE THREAD

**Issue by edwardpmorris** [gee-community/ee-runner#15]
Trying to run a pipeline that completes without error in JS Playground via ee-runner and I get this error;

```javascript
{ func: 
   { signature_: 
      { args: [Array],
        description: 'Applies a filter to a given collection.',
        returns: 'FeatureCollection',
        type: 'Algorithm',
        hidden: false,
        name: 'Collection.filter' } },
  args: 
   { collection: { func: [Object], args: [Object], varName: null },
     filter: 
      { func: [Object],
        args: [Object],
        varName: null,
        filter_: [Array] } },
  varName: null }
/builds/edwardpmorris/ee-runner-ci-cti-coal-plants/src/modules/getImageCollections.js:120
  var results = ee.Join.saveAll('matches').apply(distinct, images, filter);
                        ^

TypeError: Cannot read property 'saveAll' of undefined
    at mosaicByTime (/builds/edwardpmorris/ee-runner-ci-cti-coal-plants/src/modules/getImageCollections.js:120:25)
    at Object.getOpticalCollection (/builds/edwardpmorris/ee-runner-ci-cti-coal-plants/src/modules/getImageCollections.js:207:8)
    at my_process (/builds/edwardpmorris/ee-runner-ci-cti-coal-plants/src/cti-power-plants-plume-area.js:108:18)
    at Array.map (<anonymous>)
    at /builds/edwardpmorris/ee-runner-ci-cti-coal-plants/src/cti-power-plants-plume-area.js:93:6
    at /builds/edwardpmorris/ee-runner-ci-cti-coal-plants/node_modules/ee-runner/authenticate.js:27:13
    at /builds/edwardpmorris/ee-runner-ci-cti-coal-plants/node_modules/google-auth-library/lib/auth/oauth2client.js:210:7
    at /builds/edwardpmorris/ee-runner-ci-cti-coal-plants/node_modules/google-auth-library/lib/auth/oauth2client.js:185:5
    at Request._callback (/builds/edwardpmorris/ee-runner-ci-cti-coal-plants/node_modules/google-auth-library/lib/transporters.js:106:7)
    at Request.self.callback (/builds/edwardpmorris/ee-runner-ci-cti-coal-plants/node_modules/google-auth-library/node_modules/request/request.js:186:22)
```

The offending function seems to be:

```javascript
/***
 * Sentinel-2 produces multiple images, resultsing sometimes 4x more images than the actual size.
 * This is bad for any statistical analysis.
 * 
 * by Genna
 *
 * This function mosaics images by time.
 */

var mosaicByTime = function (images, aoi) {
  var TIME_FIELD = 'system:time_start';

  var distinct = images.distinct([TIME_FIELD]);

  var filter = ee.Filter.equals({ leftField: TIME_FIELD, rightField: TIME_FIELD });
  var results = ee.Join.saveAll('matches').apply(distinct, images, filter);

  // mosaic
  results = results.map(function(i) {
    var mosaic = ee.ImageCollection.fromImages(i.get('matches'))
        .sort('system:index')
        .mosaic()
        .clip(aoi);

    return mosaic.copyProperties(i).set(TIME_FIELD, i.get(TIME_FIELD));
  });

  return ee.ImageCollection(results);
};
exports.mosaicByTime = mosaicByTime;
```

I tried looking at the [ee API change log](https://developers.google.com/earth-engine/changelog) and did not see anything related to ee.Join. Also tried looking through the API codebase, but did not locate ee.Join. Could there be a mismatch in ee API versions between ee-runner and JS playground?

**Comment by gena** [comment]
I think this has to do with some functions which are not available directly in the API, but pulled on-the-fly, probably some initialization is missing, I will investigate.

**Comment by gena** [comment]
It seems to be a bug in EE API when running under Node.js, some classes are not initialized properly, will be fixed.

**Comment by gena** [comment]
Should be fixed in 0.0.15
